osu.Game/Overlays/OverlayRulesetTabItem.cs

```diff
@@ -0,0 +1,100 @@
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets;
+using osuTK.Graphics;
+using osuTK;
+using osu.Framework.Bindables;
+
+namespace osu.Game.Overlays
+{
+    public class OverlayRulesetTabItem : TabItem<RulesetInfo>, IHasAccentColour
+    {
+        protected readonly OsuSpriteText Text;
+        private readonly FillFlowContainer content;
+
+        public override bool PropagatePositionalInputSubTree => Enabled.Value && !Active.Value && base.PropagatePositionalInputSubTree;
+
+        private readonly Bindable<Color4> accentColour = new Bindable<Color4>();
+        private readonly Bindable<Color4> currentColour = new Bindable<Color4>();
+
+        public Color4 AccentColour
+        {
+            get => accentColour.Value;
+            set => accentColour.Value = value;
+        }
+
+        protected override Container<Drawable> Content => content;
+
+        public OverlayRulesetTabItem(RulesetInfo value)
+            : base(value)
+        {
+            AutoSizeAxes = Axes.Both;
+
+            AddRangeInternal(new Drawable[]
+            {
+                content = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Direction = FillDirection.Horizontal,
+                    Spacing = new Vector2(3, 0),
+                    Child = Text = new OsuSpriteText
+                    {
+                        Origin = Anchor.Centre,
+                        Anchor = Anchor.Centre,
+                        Text = value.Name,
+                    }
+                },
+                new HoverClickSounds()
+            });
+
+            Enabled.Value = true;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            currentColour.BindValueChanged(OnCurrentColourChanged);
+            accentColour.BindValueChanged(_ => updateState());
+            Enabled.BindValueChanged(_ => updateState(), true);
+        }
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            base.OnHover(e);
+            updateState();
+            return true;
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            base.OnHoverLost(e);
+            updateState();
+        }
+
+        protected override void OnActivated() => updateState();
+
+        protected override void OnDeactivated() => updateState();
+
+        private void updateState()
+        {
+            Text.Font = Text.Font.With(weight: Active.Value ? FontWeight.Bold : FontWeight.Medium);
+
+            currentColour.Value = IsHovered || Active.Value
```

Comment:
you can't write statements like this. it's impossible to read. split it out more.